### ISSUE TITLE
Q: How to correctly implement a home-manager?

### ISSUE THREAD

**Issue by Kreyren** [hercules-ci/flake-parts#212]
I am struggling implementing flake-parts to work as a part of nixosSystem declaration can you help? So that on system rebuild it also deploys the home-manager configuration for the user, so far i figured out this:

![image](https://github.com/hercules-ci/flake-parts/assets/11302521/123ee6ba-290f-4e12-b5e1-7eea77fc0c17)

https://github.com/Kreyren/nixos-config/pull/1/files#diff-5edef9e1a197551f68e55e21f7a097cd12ec03b7f6c4ab904b01ec1de4b72330R30-R55

Which kinda defeats the purpose of flake-parts as i want to get rid of hard-coded paths so that i can just ideally do `self.homeConfigurations."kreyren@sinnenfreude"` in modules and expect it to deploy the configuration while still being able to do e.g. `nixos-rebuild --flake 'github:kreyren/nixos-config#homeConfigurations.\"kreyren@sinnenfreude\""` and have the home-manager deployed independently from the nixosSystem

**Comment by terlar** [comment]
I don't think this has anything to do with flake-parts. It is because the `self.homeConfigurations."kreyren@sinnenfreude"` is using homeManagerConfiguration function which doesn't return a module anymore. One solution is to extract all that config into a module and just use that module in the NixOS and homeManagerConfiguration function.
